### PR TITLE
[tracker] Adding a bounding box tracker.

### PR DESCRIPTION
**Pull request by ronnyvotel**
Adding a bounding box tracker which acts as an alternative to keypoint-based tracking.

Also including a test to demonstrate usage.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/781)

### CODE REVIEW

File: pose-detection/src/calculators/keypoint_tracker.ts [hunk 1]
```diff
@@ -32,10 +32,10 @@ export class KeypointTracker extends Tracker {
 
   constructor(config: TrackerConfig) {
     super(config);
-    //TODO(ronnyvotel): validate.
-    this.keypointThreshold = config.trackerParams.keypointConfidenceThreshold;
-    this.keypointFalloff = config.trackerParams.keypointFalloff;
-    this.minNumKeyoints = config.trackerParams.minNumberOfKeypoints;
+    this.keypointThreshold =
```
Comment (ardoerlemans) [inline pose-detection/src/calculators/keypoint_tracker.ts:35]: I noticed you removed the config validation TODO. Do you want to do the validation somewhere else?

File: pose-detection/src/calculators/bounding_box_tracker.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        const iou = this.iou(pose, track);
+        return iou >= this.iouThreshold ? iou : 0.0;
+      });
+    });
+    return simMatrix;
+  }
+
+  /**
+   * Computes the intersection-over-union (IoU) between a pose and a track.
+   */
```
Comment (ardoerlemans) [inline pose-detection/src/calculators/bounding_box_tracker.ts:52]: It's clear from the description, but for completeness, maybe document the parameters as well (similar to oks() and area() in the KeypointTracker)?

**Review by ardoerlemans — APPROVED** [review]
Thanks for adding this!

**Review by lina128 — APPROVED** [review]
Very neat, thank you Ronny! Besides the new tracker addition, I really appreciate the extensive tests and the refactor, it's an example of excellent craftsmanship, thank you!

> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/781#-:-Mgfvuup3XXCHVpObl4N:b-b7yg3i)* status: :shipit: complete! 2 of 1 approvals obtained (waiting on @ronnyvotel)

**Review by ronnyvotel — COMMENTED** [review]
Thank you both!

> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs-models/781#-:-MggiTtj4Y9ILQIcZ2fw:b-srtiws)* status: :shipit: complete! 2 of 1 approvals obtained (waiting on @ardoerlemans)

___
*[pose-detection/src/calculators/bounding_box_tracker.ts, line 52 at r3](https://reviewable.io/reviews/tensorflow/tfjs-models/781#-Mgd6r6c6r7Cz7EC0go4-r3-52:-Mggj1xG6LeewasmdeTc:b-2o6qeo) ([raw file](https://github.com/tensorflow/tfjs-models/blob/ba6a8f86e19c975f956ef24f2f112036d1debf00/pose-detection/src/calculators/bounding_box_tracker.ts#L52)):*
<details><summary><i>Previously, ardoerlemans (Ard Oerlemans) wrote…</i></summary><blockquote>

It's clear from the description, but for completeness, maybe document the parameters as well (similar to oks() and area() in the KeypointTracker)?
</blockquote></details>

I added a little more detail and documentation.
___
*[pose-detection/src/calculators/keypoint_tracker.ts, line 35 at r3](https://reviewable.io/reviews/tensorflow/tfjs-models/781#-Mgd6r6c6r7Cz7EC0go2-r3-35:-Mggj5AK4aHIpV25s-8D:b-rxovc8) ([raw file](https://github.com/tensorflow/tfjs-models/blob/ba6a8f86e19c975f956ef24f2f112036d1debf00/pose-detection/src/calculators/keypoint_tracker.ts#L35)):*
<details><summary><i>Previously, ardoerlemans (Ard Oerlemans) wrote…</i></summary><blockquote>

I noticed you removed the config validation TODO. Do you want to do the validation somewhere else?
</blockquote></details>

Good call. I just expanded validateTrackerConfig() which gets called in the parent class.

**Pull request merged**